homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -62,13 +62,13 @@ async def async_setup(hass: HomeAssistant, yaml_config: ConfigType) -> bool:
 
     Adds an empty filter to hass data.
     Tries to get a filter from yaml, if present set to hass data.
-    If config is empty after getting the filter, return, otherwise emit
-    deprecated warning and pass the rest to the config flow.
     """
 
-    hass.data.setdefault(DOMAIN, {DATA_FILTER: {}})
-    if DOMAIN in yaml_config:
-        hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    if DOMAIN not in yaml_config:
+        return True
+    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN].pop(CONF_FILTER)
```

Comment:
```suggestion
    if DOMAIN in yaml_config:
        hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN].pop(CONF_FILTER)
```

Reply:
Im not sure why, but this change will cause many test to fail

FAILED tests/components/azure_data_explorer/test_init.py::test_connection[Exception] - AssertionError: assert <ConfigEntryState.NOT_LOADED: 'not_loaded'> == <ConfigEntryState.SETUP_ERROR: 'setup_error'>
 +  where <ConfigEntryState.NOT_LOADED: 'not_loaded'> = <ConfigEntry entry_id=01J016RFAJVQCCWMEQHD137DC0 version=1 domain=azure_data_explorer title=cluster state=ConfigEntryState.NOT_LOADED unique_id=None>.state
 +  and   <ConfigEntryState.SETUP_ERROR: 'setup_error'> = ConfigEntryState.SETUP_ERROR

Reply:
The tests are running successfully on my machine with the above change. Can you please recheck?

```
ha-venv@edenhaus ➜ /workspaces/core (pr/kaareseras/119089) $ pytest tests/components/azure_data_explorer
Test session starts (platform: linux, Python 3.12.3, pytest 8.2.0, pytest-sugar 1.0.0)
rootdir: /workspaces/core
configfile: pyproject.toml
plugins: mock-3.14.0, rerunfailures-14.0, typeguard-4.2.1, aiohttp-1.0.5, picked-0.5.0, requests-mock-1.12.1, respx-0.21.1, cov-5.0.0, sugar-1.0.0, timeout-2.3.1, socket-0.7.0, unordered-0.6.0, github-actions-annotate-failures-0.2.0, syrupy-4.6.1, xdist-3.6.1, asyncio-0.23.6, pytest_freezer-0.4.8, anyio-4.3.0
asyncio: mode=Mode.AUTO
collected 19 items                                                                                                                                          

 tests/components/azure_data_explorer/test_config_flow.py ✓✓✓                                                                                  16% █▋        
 tests/components/azure_data_explorer/test_init.py ✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓                                                                           100% ██████████

Results (1.12s):
      19 passed
```

Reply:
Sure thing :-)

Results (3.27s):
      11 passed
       8 failed
         - tests/components/azure_data_explorer/test_config_flow.py:15 test_config_flow
         - tests/components/azure_data_explorer/test_init.py:153 test_filter[allowlist]
         - tests/components/azure_data_explorer/test_init.py:153 test_filter[denylist]
         - tests/components/azure_data_explorer/test_init.py:153 test_filter[filtered_allowlist]
         - tests/components/azure_data_explorer/test_init.py:153 test_filter[filtered_denylist]
         - tests/components/azure_data_explorer/test_init.py:268 test_connection[KustoServiceError]
         - tests/components/azure_data_explorer/test_init.py:268 test_connection[KustoAuthenticationError]
         - tests/components/azure_data_explorer/test_init.py:268 test_connection[Exception]

and when debugging the code, I get this error:

2024-06-10 16:45:40.022 ERROR (MainThread) [homeassistant.setup] Error during setup of component azure_data_explorer
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/setup.py", line 404, in _async_setup_component
    result = await task
             ^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/azure_data_explorer/__init__.py", line 69, in async_setup
    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN].pop(CONF_FILTER)
                                     ~~~~~~~~~~~^^^^^^^^
KeyError: 'azure_data_explorer'

And the integration does not load

Reply:
There is something wrong with your dev environment. Can you please check that your are using the latest version of all requirements